lang/ruby/lib/avro/io.rb

```diff
@@ -332,18 +364,14 @@ def read_enum(writers_schema, readers_schema, decoder)
 
       def read_array(writers_schema, readers_schema, decoder)
         read_items = []
-        block_count = decoder.read_long
-        while block_count != 0
-          if block_count < 0
-            block_count = -block_count
-            block_size = decoder.read_long
-          end
-          block_count.times do
+        items_count = decoder.read_array_start
```

Comment:
Why rename this to items_count? The spec refers to this as the "block's count" so I think the original name is more clear.